src/Magnum/Platform/Sdl2Application.cpp

```diff
@@ -993,15 +979,50 @@ bool Sdl2Application::mainLoopIteration() {
             default: if(!(_flags & Flag::NoAnyEvent)) anyEvent(event);
         }
     }
+    return !(_flags & Flag::Exit);
+}
 
-    /* Tick event */
-    if(!(_flags & Flag::NoTickEvent)) tickEvent();
+bool Sdl2Application::mainLoopTickEventIteration() {
+    if(!(_flags & Flag::NoTickEvent)) {
+        tickEvent();
+        return true;
+    }
+    return false;
+}
```

Comment:
I thought about this for a bit, and personally I wouldn't even add any `mainLoopTickEventIteration()` -- keep the code directly inside `mainLoopIteration()`. 

The point of `tickEvent()` is to run something on every event loop iteration even if the screen doesn't need to be updated (such as in regular apps that refresh their window only on input events) because with the implicit event loop there's no other way to achieve that. But if you have your own event loop handling anyway, you can just call that code directly instead of implementing some virtual interface. 

That should also answer your question about what to do for Application implementations that don't have a `tickEvent()`. (Though, implementing `tickEvent()` for GLFW at least would be nice too, but that'd need adding also `setMinimalLoopPeriod()` and handling all possible interactions with draw and input events, and I don't want to force you to do all that work as well ;))